src/jit/regalloc.cpp

```diff
@@ -732,12 +721,27 @@ regNumber     Compiler::raUpdateRegStateForArg(RegState *regState, LclVarDsc *ar
                 {
                     break;
                 }
-                assert(!regState->rsIsFloat);
+                assert(regState->rsIsFloat == false);
                 regState->rsCalleeRegArgMaskLiveIn |= genRegMask(nextArgReg);
             }
         }
     }
-#endif // _TARGET_ARM_
+#ifdef _TARGET_ARM64_
+    if ((argDsc->lvOtherArgReg != REG_STK) && (argDsc->lvOtherArgReg != REG_NA))
+    {
+        assert(argDsc->lvIsMultiregStruct());
+
```

Comment:
Is this is the case of 9-16 byte struct passed in two registers?  I am wondering whether this is the right place to add this logic.  Note that since the struct is not an HFA, logic on lines 716-726 would execute first.  Next this logic would execute.  Essentially for a 9-16 byte struct passed in two registers, the following registers would be added to rsCalleeRegArgMaskLiveIn

argDsc->lvArgReg,
(argDsc->lvArgReg + 1)  i.e. the next register in sequence to lvArgReg
and argDsc->lvOtherArgReg

Is that correct?